### PR TITLE
[Cherry-Pick] Orchestrator CI/CD Scenarios P0 

### PR DESCRIPTION
**Pull request by munozemilio**

### CODE REVIEW

**Review by tsuwandy — APPROVED** [review]
Looks good, thanks Emilio!

File: packages/lu/src/parser/cross-train/cross-train.js [hunk 1]
```diff
@@ -18,28 +19,63 @@ module.exports = {
...
+            // import resolver should be capable to find implicit import files with locale, for example '[import](b.lu)' is defined in a.en-us.lu, the resolver should find b.en-us.lu
+            const foundWithLocale = typedContents.filter(content => content.id === `${path.basename(fileName, fileExt)}.${locale}`)
+            if (foundWithLocale.length > 0) {
+              found = foundWithLocale
+            } else {
+              //if no locale specified file is found, just to check whether there is file without locale matched
+              found =  typedContents.filter(content => content.id === path.basename(fileName, fileExt))
+            }
+
+            if(found.length > 0) {
```
Comment (joshgummersall) [inline packages/lu/src/parser/cross-train/cross-train.js:60]: Code formatting

File: packages/lu/src/parser/cross-train/cross-train.js [hunk 2]
```diff
@@ -18,28 +19,63 @@ module.exports = {
...
           } else if (file.filePath.endsWith(fileExtEnum.QnAFile)) {
-            importedContents.push(...await file.getFilesContent(file.filePath, fileExtEnum.QnAFile))
+            importedContents.push(...await filehelper.getFilesContent(file.filePath, fileExtEnum.QnAFile))
           }
         } else {
           const fileName = path.basename(file.filePath)
+          const updateImportedContents = async function(typedContents, fileExt) {
+            let found = []
+            // import resolver should be capable to find implicit import files with locale, for example '[import](b.lu)' is defined in a.en-us.lu, the resolver should find b.en-us.lu
+            const foundWithLocale = typedContents.filter(content => content.id === `${path.basename(fileName, fileExt)}.${locale}`)
```
Comment (joshgummersall) [inline packages/lu/src/parser/cross-train/cross-train.js:52]: You could consider grouping `typedContents` by `id` and then simply extracting results.

```javascript
const typedContentsById = typedContents.reduce((groups, content) => ({...groups, [content.id]: [...groups[content.id], content]}), {});
```

Or, if you've got `lodash`, `_.groupBy(typedContents, 'id')`.

File: packages/orchestratorlib/src/build.ts [hunk 3]
```diff
@@ -72,7 +72,7 @@ export class OrchestratorBuild {
         };
       } = {
         orchestrator: {
-          modelFolder: baseModelPath,
+          modelFolder: baseModelPath.replace(/\\/g, '/'),
```
Comment (joshgummersall) [inline packages/orchestratorlib/src/build.ts:75]: What's this for?

File: packages/orchestratorlib/src/orchestratorhelper.ts [hunk 4]
```diff
@@ -63,7 +53,11 @@ export class OrchestratorHelper {
   }
 
   public static readFile(filePath: string): string {
-    return ReadText.readSync(filePath);
+    try {
+      return ReadText.readSync(filePath);
+    } catch {
+      return '';
```
Comment (joshgummersall) [inline packages/orchestratorlib/src/orchestratorhelper.ts:59]: Might be worth returning `undefined` so you can differentiate between an empty file and one that doesn't exist.

File: packages/orchestratorlib/src/orchestratorhelper.ts [hunk 5]
```diff
@@ -395,6 +391,9 @@ export class OrchestratorHelper {
     utteranceLabelDuplicateMap: Map<string, Set<string>>,
     utteranceEntityLabelsMap: Map<string, Label[]>,
     utteranceEntityLabelDuplicateMap: Map<string, Label[]>): Promise<void> {
+    if (!luContent || luContent.length === 0) {
```
Comment (joshgummersall) [inline packages/orchestratorlib/src/orchestratorhelper.ts:394]:
```typescript
if (!luContent?.length) {
  ...
}
```

File: packages/orchestratorlib/src/orchestratorhelper.ts [hunk 6]
```diff
@@ -295,6 +288,9 @@ export class OrchestratorHelper {
           utteranceLabelDuplicateMap);
         break;
       case '.json':
+        if (filePath.endsWith(OrchestratorSettings.OrchestratorSettingsFileName)) {
```
Comment (joshgummersall) [inline packages/orchestratorlib/src/orchestratorhelper.ts:291]: A comment explaining why this is ignoring files would be good future context.

File: packages/orchestratorlib/src/orchestratorhelper.ts [hunk 7]
```diff
@@ -1123,16 +1122,16 @@ export class OrchestratorHelper {
...
-        processedFiles.push(resourceToFind);
-      }
       if (fileContent) {
         retPayload.push({
           content: fileContent,
           options: {
             id: ask.filePath,
           },
         });
+        if (!processedFiles.includes(resourceToFind)) {
```
Comment (joshgummersall) [inline packages/orchestratorlib/src/orchestratorhelper.ts:1132]: Consider using a `Set`

File: packages/orchestratorlib/src/orchestratorhelper.ts [hunk 8]
```diff
@@ -1204,7 +1206,7 @@ export class OrchestratorHelper {
         Utility.debuggingLog(`Multi language recognizer file written to ${multiRecoFileName}`);
       }
 
-      bluPaths[baseName] = snapshotFile;
+      bluPaths[baseName] = snapshotFile.replace(/\\/g, '/');
```
Comment (joshgummersall) [inline packages/orchestratorlib/src/orchestratorhelper.ts:1209]: Same question why we're replacing path characters manually.

**Pull request closed without merging**